src/rtde/rtde_client.cpp

```diff
@@ -77,6 +80,10 @@ bool RTDEClient::init()
     max_frequency_ = CB3_MAX_FREQUENCY;
   }
 
+  // Cap target frequency, defaulting to maximum
+  if (target_frequency_ <= 0.0 || target_frequency_ > max_frequency_)
```

Comment:
I think, a thrown exception on an illegal frequency would make sense instead of silently going to the max value.

Reply:
I agree.